### PR TITLE
去除alter外键约束时ob_ddl_timeout的约束，修改当前代码中遗失返回码和dead code的问题

### PR DESCRIPTION
**Pull request by wanyue-wy**
### What changes were proposed in this pull request?
当前代码将ob_ddl_timeout设置为1000s，当对于超大表进行alter操作时，添加外键约束可能会因此触发超时。该PR修改了这一问题，为alter外键约束取消了ob_ddl_timeout的限制。
同时，该PR还修改了ob_expr_generator_impl.cpp中的dead code，以及ob_create_func_resolver.cpp中丢失返回码的问题。

### How was this patch tested?
alter外键约束的代码的测试方案：
1.修改ob_ddl_timeout为5s
2.构造两张分别存有一百万记录的数据的表
3.对某一张表增加外键约束，观察是否出现超时错误

执行的sql过程如下：
MySQL [test]> alter system set _ob_ddl_timeout = '5s'; 
Query OK, 0 rows affected (0.06 sec)

MySQL [test]> set ob_query_timeout = 600000000;
Query OK, 0 rows affected (0.00 sec)

MySQL [test]> create table test (id int primary key, col int);
Query OK, 0 rows affected (0.19 sec)

MySQL [test]> create table test2 (id int primary key, col int);
Query OK, 0 rows affected (0.10 sec)

MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test fields terminated by ',';
Query OK, 999001 rows affected (54.46 sec)
Records: 1000001  Deleted: 0  Skipped: 0  Warnings: 0

MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test2 fields terminated by ',';
Query OK, 999001 rows affected (45.07 sec)
Records: 1000001  Deleted: 0  Skipped: 0  Warnings: 0

MySQL [test]> set ob_log_level=debug;
Query OK, 0 rows affected (0.00 sec)

MySQL [test]> alter table test add CONSTRAINT ID_FK FOREIGN KEY(id) REFERENCES test2(id);
Query OK, 0 rows affected (3 min 36.07 sec)

其中“alter table test add CONSTRAINT ID_FK FOREIGN KEY(id) REFERENCES test2(id);”语句执行时长为3min 36.07s，根据日志输出，check_fk_constraint_data_validity函数执行的起始时间为1631085496415058，结束时间为1631085710269822，合计执行时间为213854764，即3min 33.85s。因此，更改检测外键合法性的超时时长的方案，可以去除alter外键时ob_ddl_timeout的约束。

### Checklist

- [ ] I've run the tests to see all new and existing tests pass.
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [ ] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

File: src/sql/engine/cmd/ob_table_executor.cpp [hunk 1]
```diff
@@ -1742,6 +1742,8 @@ int ObAlterTableExecutor::check_fk_constraint_data_validity(ObExecContext& ctx,
   const ObString& origin_database_name = alter_table_schema.get_origin_database_name();
   const ObString& origin_table_name = alter_table_schema.get_origin_table_name();
   const ObTableSchema* orig_table_schema = NULL;
+  THIS_WORKER.set_timeout_ts(ObTimeUtility::current_time() + OB_MAX_USER_SPECIFIED_TIMEOUT);
+  int64_t start_time = ObTimeUtility::current_time();
```
Comment (liboyang0730) [inline src/sql/engine/cmd/ob_table_executor.cpp:1746]: const，下同。

File: src/sql/engine/cmd/ob_table_executor.cpp [hunk 2]
```diff
@@ -1777,6 +1779,9 @@ int ObAlterTableExecutor::check_fk_constraint_data_validity(ObExecContext& ctx,
       LOG_WARN("fail to check data validity by inner sql", K(ret));
     }
   }
+  
+  int64_t end_time = ObTimeUtility::current_time();
+  LOG_DEBUG("exectute time:", K(start_time), K(end_time), K(end_time-start_time));
```
Comment (liboyang0730) [inline src/sql/engine/cmd/ob_table_executor.cpp:1784]: exectute time 一般指的是一条 SQL 整体的执行时间，改成 elapsed time for check_fk_constraint_data_validity 之类的吧？

File: src/sql/code_generator/ob_expr_generator_impl.cpp [hunk 3]
```diff
@@ -954,16 +954,7 @@ int ObExprGeneratorImpl::visit_enum_set_expr(ObNonTerminalRawExpr& expr, ObExprT
   } else if (OB_ISNULL(old_op = expr.get_op()) || OB_UNLIKELY(!IS_ENUM_SET_OP(old_op->get_type()))) {
     ret = OB_ERR_UNEXPECTED;
     LOG_WARN("invalid old op", K(expr), K(ret));
-  } else {
-    ObExprTypeToStr* type_to_str = static_cast<ObExprTypeToStr*>(old_op);
-    if (OB_ISNULL(type_to_str)) {
-      ret = OB_ERR_UNEXPECTED;
-      LOG_WARN("failed to static cast ObExprOperator * to ObExprTypeToStr *", K(expr), K(ret));
-    } else if (OB_FAIL(enum_set_op->deep_copy_str_values(type_to_str->get_str_values()))) {
```
Comment (liboyang0730) [inline src/sql/code_generator/ob_expr_generator_impl.cpp:962]: 为啥把 enum_set_op->deep_copy_str_values 也给删掉了……

**Comment by liboyang0730** [comment]
> ### What changes were proposed in this pull request?
> 当前代码将ob_ddl_timeout设置为1000s，当对于超大表进行alter操作时，添加外键约束可能会因此触发超时。该PR修改了这一问题，为alter外键约束取消了ob_ddl_timeout的限制。
> 同时，该PR还修改了ob_expr_generator_impl.cpp中的dead code，以及ob_create_func_resolver.cpp中丢失返回码的问题。
> 
> ### Why are the changes needed?
> ### Will break the compatibility? How if so?
> ### Does this PR introduce any user-facing change?
> ### How was this patch tested?
> alter外键约束的代码的测试方案：
> 1.修改ob_ddl_timeout为5s
> DEF_TIME(_ob_ddl_timeout, OB_CLUSTER_PARAMETER, "5s", "[1s,)",
> "the config parameter of ddl timeout"
> "Range: [1s, +∞)",
> ObParameterAttr(Section::OBSERVER, Source::DEFAULT, EditLevel::DYNAMIC_EFFECTIVE));
> 
> 2.构造两张分别存有一百万记录的数据的表
> 3.对某一张表增加外键约束，观察是否出现超时错误
> 
> 执行的sql过程如下：
> MySQL [test]> set ob_query_timeout = 600000000;
> Query OK, 0 rows affected (0.00 sec)
> 
> MySQL [test]> create table test (id int primary key, col int);
> Query OK, 0 rows affected (0.19 sec)
> 
> MySQL [test]> create table test2 (id int primary key, col int);
> Query OK, 0 rows affected (0.09 sec)
> 
> MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test fields terminated by ',';
> Query OK, 999001 rows affected (48.10 sec)
> Records: 1000001 Deleted: 0 Skipped: 0 Warnings: 0
> 
> MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test2 fields terminated by ',';
> Query OK, 999001 rows affected (45.23 sec)
> Records: 1000001 Deleted: 0 Skipped: 0 Warnings: 0
> 
> MySQL [test]> set ob_query_timeout = 10000000;
> Query OK, 0 rows affected (0.00 sec)
> 
> MySQL [test]> alter table test add CONSTRAINT ID_FK FOREIGN KEY(id) REFERENCES test2(id);
> Query OK, 0 rows affected (3 min 38.01 sec)
> 
> ### Checklist
> * [ ]  I've run the tests to see all new and existing tests pass.
> * [ ]  If this Pull Request resolves an issue, I linked to the issue in the text above.
> * [ ]  I've informed the technical writer about the documentation change if necessary.

测试的问题：
1. _ob_ddl_timeout 是个配置项，在 sys 租户下直接通过执行 alter system set _ob_ddl_timeout = '5s'; 就好了，不用改代码的，测试的时候可以随便改这个配置项。
2. 最终 check validate 在 3 min 38.01 sec 中占的时间是多少呢？ 需要说明日志里的值和 select name, data_type, value from oceanbase.__all_sys_parameter where name = '_ob_ddl_timeout'; 的结果间的关系。

**Comment by wanyue-wy** [comment]
> > ### What changes were proposed in this pull request?
> > 当前代码将ob_ddl_timeout设置为1000s，当对于超大表进行alter操作时，添加外键约束可能会因此触发超时。该PR修改了这一问题，为alter外键约束取消了ob_ddl_timeout的限制。
> > 同时，该PR还修改了ob_expr_generator_impl.cpp中的dead code，以及ob_create_func_resolver.cpp中丢失返回码的问题。
> > ### Why are the changes needed?
> > ### Will break the compatibility? How if so?
> > ### Does this PR introduce any user-facing change?
> > ### How was this patch tested?
> > alter外键约束的代码的测试方案：
> > 1.修改ob_ddl_timeout为5s
> > DEF_TIME(_ob_ddl_timeout, OB_CLUSTER_PARAMETER, "5s", "[1s,)",
> > "the config parameter of ddl timeout"
> > "Range: [1s, +∞)",
> > ObParameterAttr(Section::OBSERVER, Source::DEFAULT, EditLevel::DYNAMIC_EFFECTIVE));
> > 2.构造两张分别存有一百万记录的数据的表
> > 3.对某一张表增加外键约束，观察是否出现超时错误
> > 执行的sql过程如下：
> > MySQL [test]> set ob_query_timeout = 600000000;
> > Query OK, 0 rows affected (0.00 sec)
> > MySQL [test]> create table test (id int primary key, col int);
> > Query OK, 0 rows affected (0.19 sec)
> > MySQL [test]> create table test2 (id int primary key, col int);
> > Query OK, 0 rows affected (0.09 sec)
> > MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test fields terminated by ',';
> > Query OK, 999001 rows affected (48.10 sec)
> > Records: 1000001 Deleted: 0 Skipped: 0 Warnings: 0
> > MySQL [test]> load data infile "/home/wanyue.wy/oceanbase/large_table.txt" into table test2 fields terminated by ',';
> > Query OK, 999001 rows affected (45.23 sec)
> > Records: 1000001 Deleted: 0 Skipped: 0 Warnings: 0
> > MySQL [test]> set ob_query_timeout = 10000000;
> > Query OK, 0 rows affected (0.00 sec)
> > MySQL [test]> alter table test add CONSTRAINT ID_FK FOREIGN KEY(id) REFERENCES test2(id);
> > Query OK, 0 rows affected (3 min 38.01 sec)
> > ### Checklist
> > 
> > * [ ]   I've run the tests to see all new and existing tests pass.
> > * [ ]   If this Pull Request resolves an issue, I linked to the issue in the text above.
> > * [ ]   I've informed the technical writer about the documentation change if necessary.
> 
> 测试的问题：
> 
> 1. _ob_ddl_timeout 是个配置项，在 sys 租户下直接通过执行 alter system set _ob_ddl_timeout = '5s'; 就好了，不用改代码的，测试的时候可以随便改这个配置项。
> 2. 最终 check validate 在 3 min 38.01 sec 中占的时间是多少呢？ 需要说明日志里的值和 select name, data_type, value from oceanbase.__all_sys_parameter where name = '_ob_ddl_timeout'; 的结果间的关系。

已更新测试的相关说明

**Review by LINxiansheng — APPROVED** [review]

**Pull request merged**